src/curve_fit.jl

```diff
@@ -29,18 +29,25 @@ function lmfit(f, g, p0::AbstractArray, wt::AbstractArray; kwargs...)
 end
 
 #for inplace f and inplace g
-function lmfit(f!, g!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; kwargs...)
-    R = OnceDifferentiable(f!, g!, p0, similar(r); inplace = true)
+function lmfit(f, g, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; inplace=false, kwargs...)
+    R = OnceDifferentiable(f, g, p0, similar(r); inplace = true)
     lmfit(R, p0, wt; kwargs...)
 end
 
 #for inplace f only
-function lmfit(f!, p0::AbstractArray, wt::AbstractArray, r::AbstractArray; autodiff = :finite, kwargs...)
-    autodiff = autodiff == :forwarddiff ? :forward : autodiff
```

Comment:
We should probably keep this line

Reply:
why? this should be handled by NLSolversBase